core/src/components/datetime/datetime.tsx

```diff
@@ -930,9 +930,40 @@ export class Datetime implements ComponentInterface {
     });
   }
 
+  private getDefaultDateParts = (value?: string | null) => {
+    let dateParts;
+
+    if (typeof value !== 'undefined') {
+      dateParts = parseDate(value);
+    } else {
+      const todayParts: DatetimeParts = parseDate(getToday());
+      dateParts = todayParts;
+
+      if (todayParts.month < this.minParts?.month && todayParts.year <= this.minParts?.year || todayParts.month > this.maxParts?.month && todayParts.year >= this.maxParts?.year) {
+        /**
+         * Today's date is either below the minimum date range or exceeds the maximum
+         * date range.
+         *
+         * Take the minimum date value to default the user at the lower bounds
+         * of the date range.
+         */
+        dateParts = {
```

Comment:
I'm not completely happy with this implementation. I think defaulting a selected date should be a feature behind a property and not default behavior. 

Regardless, with the aspect of defaulting the date; both `min` and `max` support partial datetime format. So you can have an instance of `min="2021-09"`. Our date parts need the additional pieces to render the datetime accurately.

This logic attempts to remedy that, by taking the lowest value when looking at a scenario when "today" is either before the min range or after the max range. It will apply "good defaults" for the date parts with the data we have.

For instance if you are looking at:
- Today - 12/15/2021
- Min: 09/2021
- Max 10/2021

It will set the default value at the lower range, which is 09/01/2021.

Reply:
Some libraries do not seem to do this:

Vuetify: https://codepen.io/liamdebeasi/pen/vYexwPK?editors=1010
Quasar: https://codepen.io/liamdebeasi/pen/RwLpmdG?editors=1010

It does look like Angular Material has this behavior, however: https://stackblitz.com/edit/angular-yon8wf?file=src%2Fapp%2Fdatepicker-min-max-example.ts

I think it might be confusing to have a dev enter "12/15/2021" and have the date show up as "09/01/2021" without warning. Happy to discuss more on this though.

Reply:
Tried checking what Flutter does and the datepicker seems to crash when the current value is out of range. Put the code sample from https://api.flutter.dev/flutter/material/showDatePicker.html into the DartPad demo: https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/date_pickers/regular&gh_ref=develop

From there, change the value on value/min/max dates to be what we have in this sample.